common/src/Renderer/BrushRendererArrays.h

```diff
@@ -69,26 +70,30 @@ namespace TrenchBroom {
         template<typename T>
         class VboBlockHolder {
         protected:
+            VboType m_type;
             std::vector<T> m_snapshot;
             DirtyRangeTracker m_dirtyRange;
-            VboBlock *m_block;
-
+            VboManager* m_vboManager;
+            Vbo* m_block;
         private:
             void freeBlock() {
                 if (m_block != nullptr) {
-                    m_block->free();
+                    m_vboManager->destroyVbo(m_block);
                     m_block = nullptr;
                 }
             }
 
-            void allocateBlock(Vbo &vbo) {
+            void allocateBlock(VboManager& vboManager) {
+                if (m_vboManager != nullptr) {
+                    assert(m_vboManager == &vboManager);
```

Comment:
This feels fishy. I suppose we need the VboManager reference so that we can update its bookkeeping information, but if that bookkeeping is only for statistics, I would consider removing it and not holding a reference to the VboManager here.

Reply:
Yeah, it's fishy to store the VboManager. In the long run VboManager shouldn't be stored.

I made the Vbo destructor private - to destroy one, you need to call `vboManager->destroyVbo(vbo)`. Wanted capture the idea that destroying one is an operation on the current GL context, it can't just run at any arbitrary time like most destructors, and having a VboManager passed to you means you're allowed to create or destroy a Vbo.

We're still relying on RAII with the Holder classes, which means that design idea isn't fully realized

Reply:
Ok, let's keep it as is for now then. We discussed revamping the OpenGL resource management anyway, so we will touch on this again.

Reply:
I added a TODO about this to the Holder destructor (and also in VertexArray.h)